### PR TITLE
boards/arduino-zero: add initial support

### PR DESCRIPTION
**Pull request by aabadie**
This PR is a first try to port RIOT on [arduino-zero](https://www.arduino.cc/en/Main/ArduinoBoardZero) board. The schematics are [here](https://www.arduino.cc/en/uploads/Main/ArduinoZero-schematic.pdf).

I took inspiration from the implementation of the samr21-xpro board as they share the same MCU and also the Arduino-due board for the arduino pin mapping.

The actual status is:
* it can be build
* it can be flashed on the board
* the timer doesn't not work
* the serial port is not working

Looking at the schematics, it seems that external oscillator is connected to the timer TC2 which is not available in the MCU implementation but maybe I'm wrong. Could this be the cause of problem with the serial port not working ?
I couldn't go further than this and I need a bit of help from more experienced RIOT devs.

### CODE REVIEW

**Comment by aabadie** [comment]
@PeterKietzmann, @kYc0o, I removed the WIP label as I could make it work thanks to @keestux work on Sodaq Autonomo.
I'm not 100% sure of the configuration (especially MUX macros) I set but the 2 UARTs are working. I didn't test SPI and I2C peripherals.

**Comment by aabadie** [comment]
For the record, I added a page in the wiki [here](https://github.com/RIOT-OS/RIOT/wiki/Board%3A-arduino-zero)

**Comment by keestux** [comment]
I'll have a look.
First quick remark. The Arduino Zero has a SAMD21G18A. The wiki has J18

File: boards/arduino-zero/include/periph_conf.h [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+ */
+#define TIMER_NUMOF         (2U)
+#define TIMER_0_EN          1
+#define TIMER_1_EN          1
+
+/* Timer 0 configuration */
+#define TIMER_0_DEV         TC3->COUNT16
+#define TIMER_0_CHANNELS    2
+#define TIMER_0_MAX_VALUE   (0xffff)
+#define TIMER_O_ISR         isr_tc3
```
Comment (keestux) [inline boards/arduino-zero/include/periph_conf.h]: Isn't this letter O (in TIMER_O_ISR) confusing? Why not zero?

Comment (aabadie) [inline boards/arduino-zero/include/periph_conf.h]: it's typo... fixed. Good catch !

**Comment by aabadie** [comment]
>  The Arduino Zero has a SAMD21G18A. The wiki has J18

I can't find it. Did you fix it ?

**Comment by keestux** [comment]
Yeah, it was in a table further below. I fixed it.

File: boards/arduino-zero/include/periph_conf.h (hunk 1 above)
Comment (keestux) [inline boards/arduino-zero/include/periph_conf.h]: I diffed with Autonomo, and then it popped up.
BTW I'll probably (given enough time) will use your Arduino pin stuff and adapt it for Autonomo.

**Comment by keestux** [comment]
Instead of "trying to make murdock happy", the commit comment (e0dcaa3) should be something like: "remove leading zero to avoid interpretation as octal numbers"

**Comment by aabadie** [comment]
@keestux, you are refering an old commit I squashed while I was fixing errors raised by Murdock. I think that in the end, there will be only one commit to merge in this PR.

**Comment by aabadie** [comment]
The failing test on Murdock doesn't seem related.
Can someone give a positive review ? @kYc0o, @keestux

**Comment by aabadie** [comment]
> It looks OK to me

Thanks for taking the time to review @keestux.
Now we have to wait for a maintainer (maybe @PeterKietzmann, @thomaseichinger or @kYc0o) to approve this PR, e.g give an ACK.

**Comment by aabadie** [comment]
Is there someone around that could give an ACK (and an approved review) ?

**Comment by kYc0o** [comment]
In your first text you said that timers and serial port are not working. Is that still true?

**Comment by aabadie** [comment]
@kYc0o, according to the [second comment](https://github.com/RIOT-OS/RIOT/pull/5475#issuecomment-250722638), I validated 2 serial ports. I still need to validate the other features.

**Comment by aabadie** [comment]
@kYc0o, I found the other Arduino M0 if you to try it.

**Comment by kYc0o** [comment]
Oh yeah, I didn't see the message. It's ok, so let me know if the timers work (and especially xtimer) and I'll try to get a board to test.

**Comment by kYc0o** [comment]
And go!

**Pull request merged**